sda/cmd/api/api.go

```diff
@@ -283,6 +290,63 @@ func ingestFile(c *gin.Context) {
 	c.Status(http.StatusOK)
 }
 
+// The deleteFile function deletes files from the inbox and marks them as
+// discarded in the db. Files are identified by their ids and the user id.
+func deleteFile(c *gin.Context) {
+
+	inbox, err := storage.NewBackend(Conf.Inbox)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	submissionUser := c.Param("username")
+	log.Debug("submission user:", submissionUser)
+
+	fileID := c.Param("file")
+	fileID = strings.TrimPrefix(fileID, "/")
+	log.Debug("submission file:", fileID)
+	if fileID == "" {
+		c.AbortWithStatusJSON(http.StatusBadRequest, "file ID is required")
+	}
+
+	filePath := ""
+	// Get the file path from the fileID and submission user
+	if filePath, err = Conf.API.DB.GetInboxFilePathFromID(submissionUser, fileID); err != nil {
+		log.Errorf("getting file from fileID failed, reason: (%v)", err)
+		c.AbortWithStatusJSON(http.StatusNotFound, "File could not be found in inbox")
+
+		return
+	}
+
+	// Requires a filepath instead of fileID
+	// Note: The remove fails randomly sometimes
```

Comment:
```suggestion
```

File removal should never fail randomly.
If this is in the tests it's  because of timing issues in the S3 backend (metadata not stored before deletion is initated).

Reply:
This happened to us during manual testing, not the integration tests. I have no idea why.  That's why we had to put the retry loop here. We could remove the comment, of course, but unless someone has a good idea of why this happened and why it will never happen again, the loop would be good to keep.

Reply:
Knowing the error would make it easier to debug.
But as stated above it's mot likely a timing issue.